### DISCUSSION TITLE
How are rotor diameters assigned to turbines?

### DISCUSSION THREAD

**Discussion by henrysanders3** [NatLabRockies/floris#595] [Q&A] [answered]
Hi all,

In this line of the [example code #10](https://github.com/NREL/floris/blob/main/examples/10_opt_yaw_single_ws.py), which file is it reading the rotor diameters from?

```python
print(fi.floris.farm.rotor_diameters)
```

many thanks

**Comment by rafmudaf** [comment, marked as answer]
The rotor diameter is set in the input file. See https://github.com/NREL/floris/blob/main/examples/10_opt_yaw_single_ws.py#L29. In this case, it's using one of the turbines in the included turbine library, the NREL 5MW. That turbine definition is [here](https://github.com/NREL/floris/blob/main/floris/turbine_library/nrel_5MW.yaml).

To change the rotor diameter, you can either create an external turbine library (see #568 for a description of this feature and usage) or modify the rotor diameter within your script.

**Reply by henrysanders3** [reply]
HI @rafmudaf, I have altered the rotor diameter in the the turbine library, however when the line below is run, it still prints the value 126, despite it being changed to 107.

print(fi.floris.farm.rotor_diameters)

**Reply by rafmudaf** [reply]
I think @Bartdoekemeijer's script below adds the clarity I left out, and it should address your question. However, it's worth trying to understand why modifying the rotor diameter in the turbine library yaml files directly didn't work for you. @henrysanders3 how did you install floris?

**Comment by Bartdoekemeijer** [comment]
The answer by @rafmudaf is by far the most complete and correct solution. In addition, here's a quick way to update rotor diameters in a script-like fashion:

```
# Load default model and show rotor diameters
fi = FlorisInterface("inputs/gch.yaml") # GCH model matched to the default "legacy_gauss" of V2
print(fi.floris.farm.rotor_diameters)

# Update rotor diameters manually
turbine_dict = fi.floris.farm.turbine_definitions[0]  # Copy properties of first turbine
turbine_dict["rotor_diameter"] = 107.0  # Change rotor diameter
turbine_dicts = [turbine_dict] * fi.floris.farm.n_turbines  # Copy these turbine properties over to make a list with length equal to the number of wind turbines in the farm
fi.reinitialize(turbine_type=turbine_dicts)  # Update FLORIS object with these new turbine definitions
print(fi.floris.farm.rotor_diameters)
```

**Reply by henrysanders3** [reply]
thats really helpful, many thanks
